src/extension/utils/confFileToFormData.ts

```diff
@@ -0,0 +1,372 @@
+        const type = '.' + labelTypeArr[1]
+        const fileInFormat = {
+          value: specFile,
+          label: label,
+          path: path,
+          type: type,
+        }
+        form.specObj.specFile = fileInFormat
+      }
+    }
+  }
+
+  processSolidityAttributes(confFile, form.solidityObj)
+  processSpecAttributes(confFile, form.specObj)
+
+  if (confFile.msg) {
+    form.verificationMessage = confFile.msg
+  }
+
+  // additional contracts
+  if (confFile.files && confFile.files.length > 1) {
+    processAdditionalContracts(confFile, form)
+  }
+  return form
+}
+
+function getContractNameFromFile(fileStr: string): string {
+  const contract = fileStr.includes(':')
+    ? fileStr.split(':').reverse()[0]
+    : fileStr.split('/').reverse()[0].replace('.sol', '')
+  return contract
+}
+
+/**
+ * fill additional contracts values from the conf file
+ * @param confFile to get values from (solidity file, contract name, compiler, link)
+ * @param form to fill additional contract data
+ */
+function processAdditionalContracts(confFile: ConfFile, form: NewForm): void {
+  const tempFormArr: SolidityObj[] = []
+  confFile.files?.forEach(contractStr => {
+    const mainContract = getContractNameFromFile(contractStr)
+    if (mainContract !== form.solidityObj.mainContract) {
+      // create contract
+      const tempForm: SolidityObj = {
+        mainFile: '',
+        mainContract: '',
+        linking: [],
+        compiler: {
+          exe: '',
+          ver: '',
+        },
+        solidityArgs: [],
+        solidityPackageDefaultPath: '',
+        solidityPackageDir: [],
+      }
+
+      tempForm.mainFile = contractStr as string
+      if (tempForm.mainFile.includes(':')) {
+        const contractArr = contractStr.split(':')
+        tempForm.mainFile = contractArr[0]
+        tempForm.mainContract = contractArr[1]
+      }
+      // maybe main contract isn't explicitly mentioned
+      if (!tempForm.mainContract) {
+        const splittedPathToFile = tempForm.mainFile.split('/')
+        tempForm.mainContract = splittedPathToFile[
+          splittedPathToFile.length - 1
+        ].replace('.sol', '')
+      }
+      // link
+      if (confFile.link && confFile.link.length > 0) {
+        processLink(confFile.link, tempForm)
+      }
+
+      // solc map:
+      if (confFile.solc_map) {
+        Object.entries(confFile.solc_map).forEach(([key, value]) => {
+          if (key === tempForm.mainContract) {
+            processCompiler(value, tempForm)
+          } else if (key === form.solidityObj.mainContract) {
+            processCompiler(value, form.solidityObj)
+          }
+        })
+      } else if (confFile.solc) {
+        if (confFile.solc.includes('/')) {
+          const solcArr = confFile.solc.split('/')
+          tempForm.compiler.ver = solcArr.pop() || ''
+          tempForm.compiler.exe = solcArr.reverse().join('/')
+        } else {
+          tempForm.compiler.ver = confFile.solc
+        }
+      }
```

Comment:
else if (confFile.solc?.includes('/')) {
          const solcArr = confFile.solc.split('/')
          tempForm.compiler.ver = solcArr.pop() || ''
          tempForm.compiler.exe = solcArr.reverse().join('/')
        } else {
          tempForm.compiler.ver = confFile.solc
      }

      tempForm.compiler.exe = solcArr.reverse().join('/') - this is why reverse() is so bad to get the last member in a string/array